assets/training/general/environments/acpt-pytorch-2.0-cuda11.7/tests/pytorch2_0_sample_test.py

```diff
@@ -41,6 +41,7 @@ def test_pytorch_2_0():
     job = command(
         code=this_dir / JOB_SOURCE_CODE,  # local path where the code is stored
         command="pip install -r requirements.txt" \
+                " && export DS_ACCELERATOR='cuda'"\
```

Comment:
When the accelerator is not explicitly set, `DeepSpeed` will cycle through different accelerators. It will do so by checking whether certain packages are installed (`intel_extension_for_deepspeed`, `intel_extension_for_pytorch`, `torch_npu`, `torch.mps`) and setting the accelerator that corresponds to the package (`xpu`, `cpu`, `npu`, `mps`). Only after it cycles through all of these will it set `cuda`. This is not a problem if none of those packages are installed, but since we installed `intel_extension_for_pytorch`, it will try to set the accelerator to `cpu`. This was problematic for the `gpu` smoke test since it wouldn't use the correct hardware. See [real_accelerator.py](https://github.com/microsoft/DeepSpeed/blob/master/accelerator/real_accelerator.py#L108) for reference.

Setting `DS_ACCELERATOR='cuda'` forces DeepSpeed to use the `cuda` and not `cpu`

Reply:
So any customer using acpt image has to explicitly set ds_accelerator to cuda?

Reply:
As discussed offline, I would not recommend checkin this PR unless we have Deepspeed default to GPU and only enable IPEX when asked explicitly.

Reply:
listing as resolved as they are outdated. will follow up on this issue in comment chain